src/Stratis.Bitcoin.Features.Consensus/CoinViews/DBreezeCoinView.cs

```diff
@@ -213,15 +215,25 @@ public Task SaveChangesAsync(IEnumerable<UnspentOutputs> unspentOutputs, IEnumer
 
                         this.SetBlockHash(transaction, nextBlockHash);
 
-                        all.Sort(UnspentOutputsComparer.Instance);
-                        foreach (UnspentOutputs coin in all)
+                        var toRemove = all.Where(utxo => utxo.IsPrunable == true).ToList();
```

Comment:
I suggest this could even be made better if you iterate over `all` only once by creating two lists in advance.
Then sort them independently.

Reply:
Yes it's true but very negligible because those are few ms per SaveChangesAsyn code (it's not in a loop) and keep the code compact.

I'd be worried more about using `IEnumerable<UnspentOutputs>` as a signature for this method, because we are always passing lists and arrays so we should use `IReadOnlyCollection` here, I already tried changed it in another local branch and it requires only to change the signature on classes that implement this method, no need to change anything else and the pros is that now you can handle `unspentOutputs` as a List (with Count property too) and no need to call `all = unspentOutputs.ToList()`

I could then change the code to have only an iteration on it, splitting items on these 2 arrays

Reply:
ok sure this sounds like a good idea, we anyway just use lists so if you want to add this change to this PR fine with me.

Reply:
For example RewindData is a list so the other components should be the same

Reply:
>I'd be worried more about using IEnumerable as a signature for this method
What i mean here is that when we passed IEnumerable you did a .ToList() that just copied the passed list (because it's already a list or an array) so we were doing something unuseful.

About the iteration, I could optimize them doing these steps:
- changing the signature to be `IReadOnlyCollection`
- removing `List<UnspentOutputs> all = unspentOutputs.ToList();` because I've already a list (or better a IReadOnlyCollection)
- sort all the items
- iterate through all items in the list, checking if `IsPrunable` is true: if true, i remove the item, if false, I add the item to a `toInsert` list

at this point, i can iterate the toInsert list, that will be a subset of the initial list, to insert what I have to.

Total iterations: 1 and a half (or better 1 full iteration + 1 partial iteration of only `toInsert` items :)

Reply:
Having said this, I don't think it's important how we create these lists because the impact is few ms per method call, and this method is called once every 30 or 60 seconds I don't remember.
The game changer, performance wise, on this method was the mixing of removal and insertion of items in a single loop and the `Technical_SetTable_OverwriteIsNotAllowed` instruction.